www/source/blog/2018-05-11-changes-in-the-0.56.0-supervisor.html.md

```diff
@@ -0,0 +1,64 @@
+
+Before the 0.56.0 release, the communication protocol between the `hab` commandline tool and a running Supervisor essentially consisted of modifying files on disk. While this is a very simple interaction model to implement and think about, it leaves much to be desired. It is difficult to achieve synchronous interactions between process. It is difficult to validate interactions and abort them in the face of errors. It has introduced several difficult-to-reason-about edge-case bugs. It is inherently restricte to "same machine" interaction patterns. For these reasons and more, we needed to devise a new communication protocol.
+
+With the 0.56.0 release, we have transitioned away from file-based communication to a defined TCP protocol between `hab` and a Supervisor process. Defining a protocol brings added rigor to the Supervisor codebase, which brings additional safety, enhanced backward-compatibility between releases going forward, and lays the groundwork for features that would otherwise have been impossible to implement. While this change has been introduced in a way that it can be largely transparent to current Habitat users in the near term, there are some manual steps that current users will need to take to ensure a smooth transition. We will highlight these changes now.
+
+## Install `core/hab/0.56.0` on _all_ Supervisor Machines
+
+Since communication with the new Supervisor takes place strictly via TCP, it needs a client that also speaks this same protocol. That client is the `hab` CLI tool. This means that in order to continue interacting with a Supervisor, you will need to manually install `core/hab/0.56.0` on all machines running the 0.56.0 Supervisor; if your Supervisors are configured to auto-update, then they will automatically begin running the 0.56.0 release, but you will still need to upgrade `core/hab`. In a forthcoming release, we will begin bundling these into a single package, but for now, you must take this manual step.
+
+## Revised Command Suite
+
+With this reworking of the interaction model of Habitat, we had the opportunity to streamline and clarify the suite of commands for interacting with services. One of the most frequently-asked questions about Habitat has been "what's the difference between `hab start` and `hab load`?". This has been difficult to clearly explain in the past, but now we have very clearly differentiated _all_ the commands, in a way that we hope will be more easy to reason about.
+
+Please review any scripts or automation you have that make use of these commands and update them as necessary.
+
+`hab svc load` is how you inform a Supervisor to manage a new service. This will also start the service once it has been loaded.
+
+`hab svc unload` is how you remove a service from the Supervisor. If the service is currently running, it will be stopped before being unloaded.
+
+`hab svc start` is how you start a previously-stopped, loaded service.
```

Comment:
Will start load a service if it is not loaded?

Reply:
Additionally, if the supervisor isn't running, will it start one?

Reply:
Nope!

Reply:
Considering how common it's been to use `start` up to now, it might be helpful just to mention these; I had the same two questions when I reached the end of this line. 😄 I suspect many readers will also.